platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemId.java

```diff
@@ -48,6 +50,27 @@ public ProblemGroup getGroup() {
         return parent;
     }
 
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
```

Comment:
This is not for testing purposes.
This is a new functionality to centralize how an ID should be displayed.

This functionality can be shared between the different rendering platforms (console, HTML reports, etc...) to provide a simple yet unified look to the IDs.

I will consider moving the solution to a recursive variant. I tried to avoid an additional method.